File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/SharedPreferenceUtils.java
package com.remotearthsolutions.expensetracker.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class SharedPreferenceUtils {

    private static SharedPreferenceUtils instance;
    private SharedPreferences sp;
    private SharedPreferences.Editor editor;

    private SharedPreferenceUtils(Context context) {
        sp = PreferenceManager.getDefaultSharedPreferences(context);
        editor = sp.edit();
    }

    public static SharedPreferenceUtils getInstance(Context context) {
        if (instance == null) {
            instance = new SharedPreferenceUtils(context);
        }

        return instance;
    }

    public void putInt(String key, int val) {
        editor.putInt(key, val).apply();
    }

    public int getInt(String key, int defaultVal) {
        return sp.getInt(key, defaultVal);
    }

    public void putString(String key, String val) {
        editor.putString(key, val).apply();
    }

    public String getString(String key, String defaultVal) {
        return sp.getString(key, defaultVal);
    }


    public void putBoolean(String key, boolean val) {
        editor.putBoolean(key, val).apply();
    }

    public boolean getBoolean(String key, boolean defaultVal) {
        return sp.getBoolean(key, defaultVal);
    }

    public void logOut() {
        editor.clear().commit();

    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/SharedPreferenceUtils.kt
package com.remotearthsolutions.expensetracker.utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SharedPreferenceUtils private constructor(context: Context) {

    private val sp: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
    private val editor: SharedPreferences.Editor
    fun putInt(key: String?, value: Int) {
        editor.putInt(key, value).apply()
    }

    fun getInt(key: String?, defaultVal: Int): Int {
        return sp.getInt(key, defaultVal)
    }

    fun putString(key: String?, value: String?) {
        editor.putString(key, value).apply()
    }

    fun getString(key: String, defaultVal: String): String {
        return sp.getString(key, defaultVal)
    }

    fun putBoolean(key: String?, value: Boolean) {
        editor.putBoolean(key, value).apply()
    }

    fun getBoolean(key: String?, defaultVal: Boolean): Boolean {
        return sp.getBoolean(key, defaultVal)
    }

    fun clear() {
        editor.clear().commit()
    }

    companion object {
        private var instance: SharedPreferenceUtils? = null

        fun getInstance(context: Context): SharedPreferenceUtils? {
            if (instance == null) {
                instance = SharedPreferenceUtils(context)
            }
            return instance
        }
    }

    init {
        editor = sp.edit()
    }
}